src/Spk.Common.Helpers/Service/ServiceResult.cs

```diff
@@ -21,10 +24,23 @@ public string GetFirstError()
         {
             return Errors.FirstOrDefault();
         }
+
+        public void AddWarning(string warning)
+        {
+            warning.GuardIsNotNull(nameof(warning));
```

Comment:
`GuardIsNotNullOrWhiteSpace` would be a bit better here. Should be done in `AddError` too

Reply:
@thatfrankdev Good call. It's done.